File: backend/schemas/message.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class MessageCreate(BaseModel):
    content: str
    choice_index: Optional[int] = None


class QuestionData(BaseModel):
    question_text: str
    question_type: str  # single_choice | multiple_choice | free_text | yes_no | scale
    choices: list[str] = []
    category: str
    reasoning: str


class MessageResponse(BaseModel):
    id: str
    session_id: str
    sequence: int
    role: str
    content: str
    structured_data: Optional[dict] = None
    node_id: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationResponse(BaseModel):
    message: MessageResponse
    question: Optional[QuestionData] = None
    diagram_updates: int = 0

File: frontend/src/types/message.ts
export interface Message {
  id: string;
  session_id: string;
  sequence: number;
  role: "user" | "assistant" | "system";
  content: string;
  structured_data: QuestionData | null;
  node_id: string | null;
  created_at: string;
}

export interface QuestionData {
  question_text: string;
  question_type:
    | "single_choice"
    | "multiple_choice"
    | "free_text"
    | "yes_no"
    | "scale";
  choices: string[];
  category: string;
  reasoning: string;
}

export interface ConversationResponse {
  message: Message;
  question: QuestionData | null;
  diagram_updates: number;
}
